### PR TITLE
HEXXLE-80 HEXXLE-81 Implemented info section and pause screen

### PR DESCRIPTION
**Pull request by DennisBannerman**

### CODE REVIEW

File: unity/Assets/Scripts/Unity/UnityMap.cs [hunk 1]
```diff
@@ -33,27 +35,32 @@ private void Awake()
 
         private void Start()
         {
-            unityStack = GameObject.FindGameObjectWithTag("Stack").GetComponent<UnityStack>();
-            unityPoints = GameObject.FindGameObjectWithTag("Points").GetComponent<UnityPoints>();
+            unityStack = GameObjectFinder.Stack.GetComponent<UnityStack>();
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Unity/UnityMap.cs:38]: Why was this changed again, this was recently changed to `FindGameObjectWithTag`

**Review by RicardoMonteiroSimoes — COMMENTED** [review]
Overall looks good, wasn't able to test it, would be happy if one of the other reviewers could check it out

**Comment by denadex** [comment]
Just glanced over the code looks good to me.
But I noticed a bug while testing:
Start Game -> ESC into PauseMenu -> Exit to MainMenu -> Start a new Game -> ESC-Button doesn't work anymore for PauseMenu.... (The Menu-Button still works tho)
I'm not sure if these changes caused that problem... but all in all looks good!

**Review by denadex — APPROVED** [review]
Comment above... First time reviewing :)

File: unity/Assets/Scripts/Unity/UnityMap.cs (hunk 1 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityMap.cs:38]: I ended up using the previous, regular FindGameObjectWithTag statements multiple times which is error prone and quickly becomes a hassle to change. With the new central way of defining it once in the GameObjectFinder class, we can change whatever we like once and have it be consistent across any usage case.

**Comment by DennisBannerman** [comment]
> 
> 
> Just glanced over the code looks good to me.
> But I noticed a bug while testing:
> Start Game -> ESC into PauseMenu -> Exit to MainMenu -> Start a new Game -> ESC-Button doesn't work anymore for PauseMenu.... (The Menu-Button still works tho)
> I'm not sure if these changes caused that problem... but all in all looks good!

Good catch! I think this may have been caused by the EscapeEventHandler not being enabled/disabled correctly which caused the handling method to be added as a subscriber every time EscapeEventHandler's Awake method was called. So if you did your procedure an odd number of times, things work would work; even number -> not so much.

**Comment by RicardoMonteiroSimoes** [comment]
I just tested it on my machine - it seems to work. What I noticed is that there when using "ESC" to get out of the pause menu, the current state is "saved". As in:
- Press ESC -> Pause Menu opens
- Select Info Section -> Info Section opens
- Press ESC -> Pause Menu + Info Section closes
- Press ESC -> Info section is already opened

Doesn't seem like a bug to me, just a certain behaviour, if this is fine for you, its fine for me, one could argue its a faster way of comparing tile rules with the current board 😉

**Comment by deSaoJose** [comment]
> 
> 
> I just tested it on my machine - it seems to work. What I noticed is that there when using "ESC" to get out of the pause menu, the current state is "saved". As in:
> 
>     * Press ESC -> Pause Menu opens
> 
>     * Select Info Section -> Info Section opens
> 
>     * Press ESC -> Pause Menu + Info Section closes
> 
>     * Press ESC -> Info section is already opened
> 
> 
> Doesn't seem like a bug to me, just a certain behaviour, if this is fine for you, its fine for me, one could argue its a faster way of comparing tile rules with the current board 😉

I could try to change this, once its merged, since my task was done magically xD
Apart from this, everything seems to work. It's not the must beautiful menu/info but this is a problem for the future.

**Review by deSaoJose — APPROVED** [review]
Looks Good

**Review by N2Drew — APPROVED** [review]
Neato :)
